casper/protocols/blockchain/blockchain_plot_tool.py

```diff
@@ -17,23 +17,22 @@ def __init__(self, display, save, view, validator_set):
 
         self.blockchain = []
         self.communications = []
+
         self.block_fault_tolerance = {}
         self.message_labels = {}
+        self.justifications = {
+            validator: []
+            for validator in validator_set
+        }
 
         self.message_labels[self.genesis_block] = "G"
 
-    def update(self, message_paths=None, sent_messages=None, new_messages=None):
+    def update(self, new_messages=None):
```

Comment:
Directional sending of messages is no longer a thing. The only thing we care about now are the new messages that were generated. The justifications of these messages implicitly tell us about the communications.